chemprop/uncertainty/uncertainty_predictor.py

```diff
@@ -136,50 +136,158 @@ def calculate_predictions(self):
+
+        return preds_new
+
+    @staticmethod
+    def make_intervals(preds):
+        """
+        Make uncalibrated intervals from the uncalibrated predictions.
+        """
+
+        return preds[:].tolist()
+
+    def calculate_predictions(self):
+        for i, (model, scaler_list) in enumerate(
+            tqdm(zip(self.models, self.scalers), total=self.num_models)
+        ):
+            (scaler, features_scaler, atom_descriptor_scaler, bond_feature_scaler) = scaler_list
+            if (
+                features_scaler is not None
+                or atom_descriptor_scaler is not None
+                or bond_feature_scaler is not None
+            ):
+                self.test_data.reset_features_and_targets()
+                if features_scaler is not None:
+                    self.test_data.normalize_features(features_scaler)
+                if atom_descriptor_scaler is not None:
+                    self.test_data.normalize_features(
+                        atom_descriptor_scaler, scale_atom_descriptors=True
+                    )
+                if bond_feature_scaler is not None:
+                    self.test_data.normalize_features(bond_feature_scaler, scale_bond_features=True)
+
+            preds = predict(
+                model=model,
+                data_loader=self.test_data_loader,
+                scaler=scaler,
+                return_unc_parameters=False,
+            )
+            if i == 0:
+                sum_preds = np.array(preds)
+                if self.individual_ensemble_predictions:
+                    individual_preds = np.expand_dims(np.array(preds), axis=-1)
+            else:
+                sum_preds += np.array(preds)
+                if self.individual_ensemble_predictions:
+                    individual_preds = np.append(
+                        individual_preds, np.expand_dims(preds, axis=-1), axis=-1
+                    )
+
+        self.uncal_preds = (sum_preds / self.num_models)
+        self.uncal_intervals = self.make_intervals(self.uncal_preds)
+        if self.individual_ensemble_predictions:
+            self.individual_preds = individual_preds.tolist()
+
+        self.uncal_preds = self.reformat_preds(self.uncal_preds).tolist()
+
+    def get_uncal_output(self):
+        return self.uncal_intervals
+
+
+class ConformalRegressionPredictor(ConformalQuantileRegressionPredictor):
+    """
+    Class that is used for basic conformal regression. Reformats preds to be midpoint
+    of interval and outputs uncalibrated interval size 0 as the uncal_output.
+    """
+
+    @staticmethod
+    def reformat_preds(preds):
+        """
+        Reformat preds so to midpoint of quantiles
+        """
+
+        return preds
+
+    @staticmethod
+    def make_intervals(preds):
+        """
+        Make uncalibrated intervals from the uncalibrated predictions.
+        """
+
+        intervals = np.concatenate((preds[:], preds[:]), axis=1)
```

Comment:
What are the shape sizes for preds and intervals here? If I understand correctly, the `preds` is in the shape of (data, 2 * tasks) and those values are lower and upper quantiles for each task in each point. If so, the return value of `reformat_preds ()` is not the midpoint of quantiles and the `intervals` in L278 is in the shape of (data, 4 * tasks) which should be (data, 2 * tasks) instead.

Reply:
The comment I had in line 972 is wrong, preds should be shape (data, tasks) and intervals is shape (data, 2 * tasks)